corehq/apps/userreports/reports/data_source.py

```diff
@@ -185,10 +188,50 @@ def get_total_records(self):
             raise TableNotFoundWarning
 
     def get_total_row(self):
-        return get_total_row(
-            self.get_data(), self.aggregation_columns, self.column_configs,
-            get_expanded_columns(self.column_configs, self.config)
-        )
+        def _clean_total_row(val, col):
+            if isinstance(val, numbers.Number):
+                return val
+            elif col.calculate_total:
+                return 0
+            return ''
+
+        expanded_columns = get_expanded_columns(self.column_configs, self.config)
+
+        def _get_relevant_column_ids(col):
+            return expanded_columns.get(col.column_id, [col.column_id])
```

Comment:
nit, but i often like to comment closures when i use them since it's not a super-common pattern in python. you could also just pass `expanded_columns` to the function